File: src/main/java/org/redrune/net/packet/context/impl/RegionLoadPacketContext.java
package org.redrune.net.packet.context.impl;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.net.packet.context.PacketContext;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
public class RegionLoadPacketContext extends PacketContext {
	
	@Override
	public void handle(Player player) {
		if (!player.getAttributes().clientHasLoadedMapRegion()) {
			player.getAttributes().setClientHasLoadedMapRegion();
		}
		player.getPackets().refreshSpawnedObjects();
		player.getPackets().refreshSpawnedItems();
	}
}

File: src/main/java/org/redrune/networking/packet/context/impl/RegionLoadPacketContext.kt
package org.redrune.networking.packet.context.impl

import org.redrune.game.entity.actor.player.Player
import org.redrune.networking.packet.context.PacketContext

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
class RegionLoadPacketContext : PacketContext() {
    override fun handle(player: Player) {
        if (!player.attributes.clientHasLoadedMapRegion()) {
            player.attributes.setClientHasLoadedMapRegion()
        }
        player.packets.refreshSpawnedObjects()
        player.packets.refreshSpawnedItems()
    }
}